app/assets/javascripts/components/group/group_create_modal.es6.jsx

```diff
@@ -23,44 +29,24 @@ class GroupCreateModal extends CreateModal {
   // --------------------------------------------------
   // Helpers
   // --------------------------------------------------
-  /** Returns a groupable object (leadership candidate) by their id, this is needed
-   * because template does not store their full name.
-   */
-  findGroupable(groupable_id) {
-    return this.props.groupables.filter((groupable) => groupable.id == groupable_id).pop();
-  }
-
-  generateChoice(groupable, leader) {
-    var primary = 0;
-    var secondary = 1;
-    switch(leader) {
-      case 'primary':
-        return {
-          action: () => ConferenceActions.storeListAttribute(
-            'leaderships_attributes',
-            primary,
-            {user_id: groupable.id}
-          ),
-          content: Helpers.humanize(groupable.full_name),
-        };
-      case 'secondary':
-        return {
-          action: () => ConferenceActions.storeListAttribute(
-            'leaderships_attributes',
-            secondary,
-            {user_id: groupable.id, style: secondary}
-          ),
-          content: Helpers.humanize(groupable.full_name),
-        };
-    }
+  createGroup() {
+    ConferenceActions.createGroup(this.props.template);
   }
 
-  generateChoices(leader) {
-    return this.props.groupables.map((groupable) => this.generateChoice(groupable, leader));
+  generateChoice(groupable, index) {
+    return {
+      action: () => ConferenceActions.storeListAttribute(
+        'leaderships_attributes',
+        { user_id: groupable.id },
+        index,
```

Comment:
I think you're using the wrong index here, (it should be 0 for a primary leader, and 1 for a secondary leader). I tested this, and, as is, you can't add any leader beyond choice 2.